src/Roave/SecurityAdvisories/Version.php

```diff
@@ -4,24 +4,36 @@
 
 namespace Roave\SecurityAdvisories;
 
-/**
- * A simple version, such as 1.0 or 1.0.0.0 or 2.0.1.3.2
- */
+use Composer\Semver\VersionParser;
+
 final class Version
 {
-    const VALIDITY_MATCHER = '/^(?:\d+\.)*\d+$/';
+    private const STABILITY_TAIL =  '[._-]?'.
+                                    '(?:(stable|beta|rc|alpha|patch)((?:[.-]?\d+)+)?)?'.
+                                    '([.-]?dev)?';
+
+    private const VALIDITY_MATCHER = '/^(?:\d+\.)*\d+'.self::STABILITY_TAIL.'$/';
+
+    /**
+     * @var string
```

Comment:
Running the patch locally, I noticed that `Version::fromString('1.0') != Version::fromString('1')`: that assumption should be restored.

In my opinion, you should have 3 separate fields here:

 * `versionNumbers` (`int[]` as before - with each value positive, mostly for ease of use)
 * `stabilityFlag` as a `StabilityFlag` (where `StabilityFlag` is an enumerable type for `stable,beta,rc,alpha,patch,dev`, field is not null)
 * `stabilityRelease` as an `int|null`

That would make it easier to reason about this object, and would make the direct comparison simpler again.